documentation/design/interactive-package-references.md

```diff
@@ -0,0 +1,217 @@
+# Packages Sourcing
+
+The feature is meant to simplify the process of fixing, testing and contributing changes in projects published as nugets.
+
+It is inspired by the golang modules design - where a standalone dependency (module) has a pointer to it's source location as a first-class citizen within the ecosystem (go.mod) and the relation between the source codes and runtime dependecy is unambigously guaranteed by the compiler.
+
+# North Star / Longer-term vision
+
+We envision the 'packages sourcing' to be a first-class-citizen within nuget client (and hence [`dotnet restore`](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-restore)). Via denoting specific metadata on `PackageReference` it would be possible to perform specific mode of restore operation for the particular package reference - by pointing to a local sources, or letting the command to figure out and fetch apropriate sources:
```

Comment:
I have a gap in understanding on the requirement/user-scenario here? Can that be updated before we talk about the north-star of the feature itself? 
For example, should developer be required to make changes to the csproj `PackageReference` or should it be a command line option to msbuild/nuget.

Reply:
/cc: @baronfel

Reply:
If we move the [User Scenarios section](https://github.com/dotnet/msbuild/pull/8475/files#diff-41d6bc3c5d881a2f2957df6c99fbabf23a7367cc08373b50eda7e971251f9847R45-R61) to the top - would that address the concern?

UX - Manual edit of `PackageReference` vs an CLI option (vs VS GUI, vs anything else) - great point! Though I would prefer avoiding a single firm decision on that now, unitl a PoC/spike (that this doc is mainly shooting for, north-star is for context) is alive and viability and limitations are better qualified and quantified